### ISSUE TITLE
Various default methods from java.util.Map don't work correctly if defRetValue is not null

### ISSUE THREAD

**Issue by KidsDontPlay** [vigna/fastutil#101]
Version: fastutil-8.2.1
```java
Object2ObjectOpenHashMap<String, String> map = new Object2ObjectOpenHashMap<String, String>();
map.defaultReturnValue("undefined");

System.out.println(map.getOrDefault("bar", "nak")); //prints 'undefined' instead of 'nak'
	
map.putIfAbsent("foo", "one"); // 'foo=>one' is not added, see next line
System.out.println(map.get("foo")); //prints 'undefined' instead of 'one'
	
System.out.println(map.replace("bar", "two")); //prints 'undefined' instead of 'null' and 'bar=>two' is added (should not be) 
	
System.out.println(map.computeIfAbsent("qux", s -> "three")); //prints 'undefined instead of 'three' and 'qux=>three is not added (should be)
	
System.out.println(map.computeIfPresent("def", (a, b) -> "four")); //prints 'four' instead of null and 'def=>four is added (should not be)
```

**Comment by vigna** [comment]
Ouch, I can see where we botched. We let Java's default implementation go through in the Object2Object case, but they don't take into account default values (which are a fastutil feature). Java's default implementation sees a non-null return value from the underlying map and returns that value.

I have to think what to do here. We might want just to document this behaviour.

**Comment by develar** [comment]
It is also performance issue. `computeIfAbsent` in java hashmap is optimized and doesn't calculate hash twice (the same reason why there is `addOrGet` method for int set).

**Comment by vigna** [comment]
This should be fixed by 82e2ad359fd09fbe8dfb7ef40a0fc5a6abb60976.
